dts/bindings/spi/spi-device.yaml

```diff
@@ -16,3 +16,7 @@ properties:
       description: Maximum clock frequency of device's SPI interface in Hz
     label:
       required: true
+    duplex:
+      type: int
+      required: false
+      description: Duplex mode, full or half. By default it's always full.
```

Comment:
I think this should use an enum (https://docs.zephyrproject.org/latest/guides/dts/bindings.html#enum-values) and a default (0), with a justification for why (that's the usual case for SPI controllers).